programs/scope/src/lib.rs

```diff
@@ -133,6 +151,50 @@ impl Default for DatedPrice {
     }
 }
 
+#[zero_copy]
+#[derive(Debug, Eq, PartialEq)]
+pub struct EmaTwap {
+    pub current_ema_1h: u128,
+
+    pub last_update_slot: u64, // the slot when the last observation was added
+    pub last_update_unix_timestamp: u64,
+
+    pub padding: [u128; 40],
+}
+
+impl Default for EmaTwap {
+    fn default() -> Self {
+        Self {
+            current_ema_1h: 0,
+            last_update_slot: 0,
+            last_update_unix_timestamp: 0,
+            padding: [0_u128; 40],
+        }
+    }
+}
+
+impl EmaTwap {
+    fn to_dated_price(&self) -> DatedPrice {
+        DatedPrice {
+            price: Decimal::from(self.current_ema_1h).into(),
```

Comment:
Careful, `current_ema_1h` is a scaled `Decimal` never use from/into on it!
```suggestion
            price: Decimal::from_scaled_val(self.current_ema_1h).into(),
```